pkg/peertls/peertls.go

```diff
@@ -110,6 +117,28 @@ func VerifyPeerCertChains(_ [][]byte, parsedChains [][]*x509.Certificate) error
 	return verifyChainSignatures(parsedChains[0])
 }
 
+// VerifyCAWhitelist verifies that the peer identity's leaf was signed by any one of the
+// (certificate authority) certificates in the provided whitelist
+func VerifyCAWhitelist(cas []*x509.Certificate) PeerCertVerificationFunc {
+	if cas == nil {
+		return nil
+	}
+
+	return func(_ [][]byte, parsedChains [][]*x509.Certificate) error {
+		var (
+			err   error
+			valid bool
+		)
+		for _, ca := range cas {
+			err = verifyCertSignature(ca, parsedChains[0][0])
+			if valid {
```

Comment:
when is valid being set to true?

Reply:
nice catch - fixed, and added some tests in bcb4f15